thrall/app/controllers/ThrallController.scala

```diff
@@ -28,4 +38,43 @@ class ThrallController(
     }
   }
 
+  implicit val pollingMaterializer:ActorMaterializer = ActorMaterializer()(actorSystem)
+
+  def startMigration = withLoginRedirectAsync {
+    val msgFailedToFetchIndex = s"Could not fetch ES index details for alias '${es.imagesMigrationAlias}'"
+    es.getIndexForAlias(es.imagesMigrationAlias) recover { case error: Throwable =>
+      logger.error(msgFailedToFetchIndex, error)
+      InternalServerError(msgFailedToFetchIndex)
+    } map {
+      case Some(index) =>
+        BadRequest(s"There is already an index '${index}' for alias '${es.imagesMigrationAlias}', and thus a migration underway.")
+      case None =>
+        messageSender.publish(CreateMigrationIndexMessage(
+          migrationStart = DateTime.now(DateTimeZone.UTC),
+          gitHash = utils.buildinfo.BuildInfo.gitCommitId
+        ))
+        // poll until images migration alias is created, giving up after 10 seconds
```

Comment:
I think this should 200 on sending the message and leave checking status for the user.

Reply:
That would require the user to refresh the page, and potentially be tempted to click the Start Migration button again (even though server-side code guards against that) OR it would require client side polling - I've tried to avoid having any client-side code for this control panel - for simplicity (to avoid the inevitable package.json, react, nvmrc etc. etc. which all needs to be kept up to date)